File: client-scalaz/src/main/scala/org/openapitools/client/api/JsonPrimitive.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import JsonPrimitive._

case class JsonPrimitive (
  asInt: Option[Integer],
number: Option[Boolean],
boolean: Option[Boolean],
asString: Option[String],
asBoolean: Option[Boolean],
asNumber: Option[Any],
asDouble: Option[Double],
asFloat: Option[Float],
asLong: Option[Long],
asByte: Option[ByteArray],
asCharacter: Option[String],
asBigDecimal: Option[BigDecimal],
asBigInteger: Option[Integer],
asShort: Option[Integer],
string: Option[Boolean],
asJsonObject: Option[JsonObject],
jsonNull: Option[Boolean],
jsonArray: Option[Boolean],
jsonObject: Option[Boolean],
jsonPrimitive: Option[Boolean],
asJsonArray: Option[JsonArray],
asJsonPrimitive: Option[JsonPrimitive],
asJsonNull: Option[JsonNull])

object JsonPrimitive {
  import DateTimeCodecs._

  implicit val JsonPrimitiveCodecJson: CodecJson[JsonPrimitive] = CodecJson.derive[JsonPrimitive]
  implicit val JsonPrimitiveDecoder: EntityDecoder[JsonPrimitive] = jsonOf[JsonPrimitive]
  implicit val JsonPrimitiveEncoder: EntityEncoder[JsonPrimitive] = jsonEncoderOf[JsonPrimitive]
}

File: client-clojure/src/sirqul_io_t_platform/specs/json_primitive.clj
(ns sirqul-io-t-platform.specs.json-primitive
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs. :refer :all]
            [sirqul-io-t-platform.specs.json-object :refer :all]
            [sirqul-io-t-platform.specs.json-array :refer :all]
            [sirqul-io-t-platform.specs.json-primitive :refer :all]
            [sirqul-io-t-platform.specs.json-null :refer :all]
            )
  (:import (java.io File)))


(def json-primitive-data
  {
   (ds/opt :asInt) int?
   (ds/opt :number) boolean?
   (ds/opt :boolean) boolean?
   (ds/opt :asString) string?
   (ds/opt :asBoolean) boolean?
   (ds/opt :asNumber) any?
   (ds/opt :asDouble) float?
   (ds/opt :asFloat) float?
   (ds/opt :asLong) int?
   (ds/opt :asByte) any?
   (ds/opt :asCharacter) string?
   (ds/opt :asBigDecimal) float?
   (ds/opt :asBigInteger) int?
   (ds/opt :asShort) int?
   (ds/opt :string) boolean?
   (ds/opt :asJsonObject) json-object-spec
   (ds/opt :jsonNull) boolean?
   (ds/opt :jsonArray) boolean?
   (ds/opt :jsonObject) boolean?
   (ds/opt :jsonPrimitive) boolean?
   (ds/opt :asJsonArray) json-array-spec
   (ds/opt :asJsonPrimitive) json-primitive-spec
   (ds/opt :asJsonNull) json-null-spec
   })

(def json-primitive-spec
  (ds/spec
    {:name ::json-primitive
     :spec json-primitive-data}))
